02-getting_started.Rmd

```diff
@@ -1,27 +1,74 @@
+
+3. If there is already a previous version of R installed in your system and you want to retain the packages installed in it, then uninstall the previous version of $\textsf{R}$ (from the Control Panel) and install the new one. In the new installation, copy any installed packages to the library folder and run `update.packages(checkBuilt=TRUE, ask=FALSE)` in the new $\textsf{R}$. 
+
+4. If there are different versions of R installed, then they are present in parallel folders. Hence, earlier versions can also be retained if you wish to keep them.
+  
+5. When there is a change in the minor version of $\textsf{R}$ (say, from 4.0.3 to 4.1.0), then the library names (folder `R\win-library\x.y` of your home directory, `R\win64-library\x.y` on 64-bit builds), will need to be updated too. For doing this, copy (say) `R\win-library\4.0` to `R\win-library\4.1` and then run the `update.packages(checkBuilt=TRUE, ask=FALSE)` command.
+
+6. Daily [tarballs](https://stat.ethz.ch/R/daily/) are available for the patched version of the current release (`r-patched`), and the development version of the current release (`r-devel`). The same are also available via the [R Subversion repository](https://svn.R-project.org/R/)
+
+## Building $\textsf{R}$ and $\textsf{R}$ packages
+
+### What tools you need to build $\textsf{R}$ from source on Windows?
+
+1. `RTools` is the [subdirectory](https://cran.r-project.org/bin/windows/Rtools/) which you want to build $\textsf{R}$ or your own $\textsf{R}$ package(s) on Windows. 
+
+2. You also need a distribution of $\LaTeX$ installed for building $\textsf{R}$ and checking packages. The `MiKTeX` distribution of $\LaTeX$ that is used on CRAN can be downloaded from https://miktex.org.
+
+### How to setup `RTools`?
+
+1. The executable version of `RTools` that is specific to your requirements can be downloaded from https://cran.r-project.org/bin/windows/Rtools/ and run in the Windows-style installer. 
+
+2. After completing the installation, you need to put the location of the RTools $\textit{make}$ $\textit{utilities}$ on the `PATH`. For doing this, exceute the following commands in $\textsf{R}$:
+
+```{r, results='hide'}
+writeLines('PATH="${RTOOLS40_HOME}\\usr\\bin;${PATH}"', con = "~/.Renviron")
+```
+
+Restart $\textsf{R}$ now. Verify that $\textit{make}$ can be found using the following command. (The output of the command should show the path where you installed `RTools`):
+
+```{r, results='hide'}
+Sys.which("make")
+## "C:\\rtools40\\usr\\bin\\make.exe"
+```
+
+If the above works, then try installing an $\textsf{R}$ package from source:
+
+```{r, results='hide'}
+## install.packages("jsonlite", type = "source")    ## An example
+```
+
+If you are successful in installing an $\textsf{R}$ package from source, then the setup for `RTools` is completed.
+
+### How to build $\textsf{R}$?
+
+To build $\textsf{R}$ for Windows using `RTools` follow the instructions in this [readme](https://github.com/r-windows/r-base#readme) file.

```

Comment:
I was not aware of these instructions. We should make it clear that only the Quick Development Build is needed here. The quick build is nice because it avoids building the manuals, which you don't generally need for development and testing. On the other hand, it does make some assumptions, e.g. that MikTeX is installed in C:/Program Files and that the user has 64-bit R (probably true for most cases). We might need to point this out and say to change the relevant lines in https://github.com/r-windows/r-base/blob/master/quick-build.sh if necessary (https://github.com/r-windows/r-base/blob/50a229fc76c50a5fb42c0daa367466aaf2318171/quick-build.sh#L6, https://github.com/r-windows/r-base/blob/50a229fc76c50a5fb42c0daa367466aaf2318171/quick-build.sh#L13). 

It might be interesting to mention that the second option is the one that is used to create the binary builds mentioned in the previous section (and not needed when building R from source yourself).